File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:

        #  2 4 1 1 1 1 1
        
        myjumps = 0
        curMaxDist = 0
        end = 0
                
        for i in range(len(nums) - 1):
            
            curMaxDist = max(curMaxDist, i + nums[i])
            
            if i == end:
                myjumps += 1
                
                end = curMaxDist
            
            
        
        return myjumps
            
            
        
        


        

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int myjumps = 0;
        int curMaxDist = 0;
        int end = 0;
                
        for(int i = 0; i < nums.size() - 1; i++){
            curMaxDist = max(curMaxDist, i + nums[i]);
            if (i == end) {
                 myjumps += 1;
                
                end = curMaxDist;
            }
        }
        
        return myjumps;
        
    }
};